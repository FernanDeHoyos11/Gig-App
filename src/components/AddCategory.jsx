import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

const [inputValue, setInputValue] = useState('SNK');
   
const onChange = (event) => {
     setInputValue(event.target.value)
     
}

const submit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
     onNewCategory( inputValue.trim())
     setInputValue('') 
}
     return(
        <form onSubmit={submit}>
            <input 
               type="text"
               placeholder="Buscar Gif"
               value={inputValue}
               onChange={onChange}
         />
        </form>
     )
}
