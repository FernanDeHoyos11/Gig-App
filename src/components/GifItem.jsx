import PropTypes  from "prop-types";
import { downloadGif } from "../helpers/download";
import { handleShare } from "../helpers/handleShare";



export const GifItem = ({ id, title, url }) => {
const handleDownload = () => {
    downloadGif(url, title);
  };
    
const Share = () => {
    handleShare(url, title)
}
    

    return (
        <div className="card">
            <p>{title}</p>
            <img src={url} alt={title} />
            <div className="div-share">
                <button className="btn-share" onClick={Share}>Compartir</button>
                <button className="btn-down" onClick={handleDownload}>Descargar</button>
            </div>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}