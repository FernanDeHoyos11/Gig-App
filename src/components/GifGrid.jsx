
import { GifItem } from "./GifItem";
import { useFetchGif } from "../hooks/useFetchGif";
import { useState } from "react";
import PropTypes  from "prop-types";

export const GifGrid = ({category}) => {
  
  const {images, isLoading} = useFetchGif(category);
  const [limit, setLimit] = useState(10);

  const moreGif = () => {
    setLimit(preLimit => preLimit +5);
  }

    return (
        <>
          <h3>{category}</h3>

          { isLoading ? (<h2>Cargando...</h2>) : null }

          <div className="card-grid">{
              images.slice(0, limit).map( (image) => (
                <GifItem
                   key={ image.id}
                   title={image.title}
                   url={image.url} 
                   // o {...image} para las props 
                />
                ))
                
              }
              
              </div>
              {images.length > limit && (
              <button className="btnShowMore" onClick={moreGif}>Ver m√°s</button>
      )}
        </>
    );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}