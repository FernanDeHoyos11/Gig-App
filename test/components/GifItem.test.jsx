import { getByText, render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('prueba en <GifItem> ', () => {
    const title = 'Saitama God';
    const url = 'https://saitama-god.png/'
    test('debe hacer match con snapshot', () => {
       const {container}  = render(<GifItem title={title} url={url} />);
       expect(container).toMatchSnapshot()
    })

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('debe mostrar titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(getByText(title)).toBeTruthy();
    })
})