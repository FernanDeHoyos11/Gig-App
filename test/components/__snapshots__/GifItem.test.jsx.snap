// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en <GifItem>  debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Saitama God
    </p>
    <img
      alt="Saitama God"
      src="https://saitama-god.png/"
    />
    <div
      class="div-share"
    >
      <button
        class="btn-share"
      >
        Compartir
      </button>
      <button
        class="btn-down"
      >
        Descargar
      </button>
    </div>
  </div>
</div>
`;
